api/v1alpha1/stresschaos_types.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Stress chaos is a chaos to generate plenty of stresses over a collection of pods.
+// A sidecar will be injected along with the target pod during creating. It's the
+// sidecar which generates stresses or cancels them. For now, we use stress-ng as
+// the stress generator for the chaos.
+
+// +kubebuilder:object:root=true
+
+// StressChaos is the Schema for the stresschaos API
+type StressChaos struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the behavior of a time chaos experiment
+	Spec StressChaosSpec `json:"spec"`
+
+	// +optional
+	// Most recently observed status of the time chaos experiment
+	Status StressChaosStatus `json:"status"`
+}
+
+// StressChaosSpec defines the desired state of StressChaos
+type StressChaosSpec struct {
+	// Mode defines the mode to run chaos action.
+	// Supported mode: one / all / fixed / fixed-percent / random-max-percent
+	Mode PodMode `json:"mode"`
+
+	// Value is required when the mode is set to `FixedPodMode` / `FixedPercentPodMod` / `RandomMaxPercentPodMod`.
+	// If `FixedPodMode`, provide an integer of pods to do chaos action.
+	// If `FixedPercentPodMod`, provide a number from 0-100 to specify the max % of pods the server can do chaos action.
+	// If `RandomMaxPercentPodMod`,  provide a number from 0-100 to specify the % of pods to do chaos action
+	// +optional
+	Value string `json:"value"`
+
+	// Selector is used to select pods that are used to inject chaos action.
+	Selector SelectorSpec `json:"selector"`
+
+	// Stressors defines plenty of stressors to be generated. It's simply a few
+	// arguments for stress-ng. For fully supported stressors, see stress-ng(1)
+	Stressors string `json:"stressors"`
+
+	// Duration represents the duration of the chaos action
+	// +optional
+	Duration *string `json:"duration"`
```

Comment:
Please add `omitempty` for `Duration` and `Scheduler`

Reply:
Thank you.  Will add it.